### PR TITLE
Another try

### PR DESCRIPTION
**Pull request by Evans-Munatsa**
- added a join table migration called menu_items
- added asociations
- edited the items controller with more simpler code
- edited the items form

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :find_menus
```
Comment (ryannealmes) [inline app/controllers/application_controller.rb]: This shouldn't be in your application controller. Think what would happen if you added a users controller (or if you have one). Every time an action is called this method would fire, resulting in an unnecessary db hit. You should put this stuff in the controller or action that needs it.

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -39,8 +41,20 @@ def destroy
...
+
   def item_params
-    params.require(:item).permit(:title, :description, :image, :menu_id, :price)
+    params.require(:item).permit(
+      [
+        :title, 
+        :description, 
+        :image, 
+        :price,
+        menu_items_attributes: [
```
Comment (ryannealmes) [inline app/controllers/items_controller.rb]: I see Mikhail helped you here. Make sure you understand what nested attributes are please.

File: app/models/menu.rb [hunk 3]
```diff
@@ -1,4 +1,6 @@
 class Menu < ActiveRecord::Base
-	belongs_to :user
-	has_many :items
+  has_many :menu_items
+  has_many :items, :through => :menu_items
+
+  # validates :name, presence: true
```
Comment (ryannealmes) [inline app/models/menu.rb]: Does menu not require a name? Should this be here?

File: spec/models/menu_item_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe MenuItem, type: :model do
```
Comment (ryannealmes) [inline spec/models/menu_item_spec.rb]: Does this spec pass?

**Comment by ryannealmes** [comment]
Overall it looks a lot better. One thing I am not sure of is if the images should be included. Don't you upload them through the front end?

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -10,4 +10,8 @@ def authenticate_admin!
       redirect_to root_path, alert: 'You are not allowed to perform that action.'
     end
   end
+
+  def find_menus
```
Comment (mdkrog) [inline app/controllers/application_controller.rb]: Where is this getting used. If it isn't I would delete it

File: app/controllers/items_controller.rb [hunk 6]
```diff
@@ -3,16 +3,17 @@ class ItemsController < ApplicationController
   before_action :authenticate_admin!, only:[ :new, :edit, :update, :create, :destroy ]
 
   def index
-    if params[:menu].blank?
+    if params[:id].blank?
```
Comment (mdkrog) [inline app/controllers/items_controller.rb]: When will the params[:id] ever not be blank?

Comment (Evans-Munatsa) [inline app/controllers/items_controller.rb]: the method (if params[:id].blank?) is to help the menu to find the items_id, however if l remove the method the whole app is going to crash, but l think l'm gonna stick with the method until l find a way to re-factor the code

**Comment by mdkrog** [comment]
Slowly getting there :+1:

**Comment by mdkrog** [comment]
:+1:

**Pull request merged**